DependencyInjection/EkinoNewRelicExtension.php

```diff
@@ -43,52 +45,65 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.xml');
 
-        if (!$config['enabled']) {
-            $config['monolog']['enabled'] = false;
-            $interactor = BlackholeInteractor::class;
-        } elseif (isset($config['interactor'])) {
-            $interactor = $config['interactor'];
-        } else {
-            // Fallback to see if the extension is loaded or not
-            $interactor = \extension_loaded('newrelic') ? NewRelicInteractor::class : BlackholeInteractor::class;
-        }
+        $container->setAlias(NewRelicInteractorInterface::class, $this->getInteractorServiceId($config));
+        $container->setAlias(TransactionNamingStrategyInterface::class, $this->getTransactionNamingServiceId($config));
 
         if ($config['logging']) {
-            $container->getDefinition(LoggingInteractorDecorator::class)
-                ->replaceArgument(0, new Reference($interactor));
-            $interactor = LoggingInteractorDecorator::class;
+            $container->register(LoggingInteractorDecorator::class)
+                 ->setDecoratedService(NewRelicInteractorInterface::class)
+                ->setArguments(
+                    [
+                        '$interactor' => new Reference(LoggingInteractorDecorator::class.'.inner'),
+                        '$logger' => new Reference('logger'),
```

Comment:
This will throw an exception if `logger` service does not exist. 

We should create a PR to make the service id for the logger configurable.

Reply:
good catch